.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,50 @@
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Build and Publish 
+
+on:
+  push:
+    branches: [ dev , master ]
+    paths-ignore: 
+      - .gradle/wrapper
```

Comment:
replicate the path changes from the other PR. And remove the push trigger from the other workflow (now I understand why you had removed it originally)